analyzers/tests/SonarAnalyzer.UnitTest/Common/AnalyzerConfigurationTests.cs

```diff
@@ -32,21 +34,88 @@ namespace SonarAnalyzer.UnitTest.Common
     public class AnalyzerConfigurationTests
     {
         [TestMethod]
-        public void GivenSonarLintXmlIsLoaded_WhenInitializeIsCalledWithDifferentPath_EnabledRulesAreUpdated()
+        public void HotspotConfiguration_WhenInitializeIsCalledWithDifferentPathAndDifferentLoaders_EnabledRulesAreUpdated()
         {
-            var sut = AnalyzerConfiguration.Hotspot;
-            sut.Initialize(new AnalyzerOptions(GetAdditionalFiles(@"a\SonarLint.xml")), GetRuleLoader("S1067"));
+            var sut = new AnalyzerConfiguration.HotspotConfiguration();
+
+            sut.Initialize(new AnalyzerOptions(GetAdditionalFiles(@"bar\SonarLint.xml")), GetRuleLoader("S1067"));
             Assert.IsTrue(sut.IsEnabled("S1067"));
+            Assert.IsFalse(sut.IsEnabled("S9999"));
 
-            sut.Initialize(new AnalyzerOptions(GetAdditionalFiles(@"b\SonarLint.xml")), GetRuleLoader("S1068"));
+            sut.Initialize(new AnalyzerOptions(GetAdditionalFiles(@"qix\SonarLint.xml")), GetRuleLoader("S9999"));
             Assert.IsFalse(sut.IsEnabled("S1067"));
-            Assert.IsTrue(sut.IsEnabled("S1068"));
+            Assert.IsTrue(sut.IsEnabled("S9999"));
+        }
+
+        [TestMethod]
+        public void HotspotConfiguration_WhenInitializeIsCalledWithTheSamePathAndDifferentLoaders_EnabledRulesAreNotUpdated()
```

Comment:
I think the rule loader is not relevant in this case since we don't have any logic related to it's equality and because of this the test names can cause confusion.

I prefer to have `HotspotConfiguration_WhenInitializeIsCalledWithDifferentPath_EnabledRulesAreUpdated` and `HotspotConfiguration_WhenInitializeIsCalledWithTheSamePath_EnabledRulesAreNotUpdated`.